backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -40,6 +46,53 @@ public Task(String identifier)
     }
 
     /**
+     * Sets the current @{@link ViewTaskActivity} for this task.
+     * This function should be called only by the {@link ViewTaskActivity} instance running
+     * at its creation time.
+     * @param activity the @{@link ViewTaskActivity} instance
+     */
+    public void setActivity(ViewTaskActivity activity){
```

Comment:
I implemented the walking tasks by extending ViewTaskActivity with more functionality.  It seemed to work well without having to store the activity in a model object like Task.

https://github.com/ResearchStack/ResearchStack/pull/349